src/Features/Core/Diagnostics/Analyzers/RudeEditUserDiagnosticAnalyzer.cs

```diff
@@ -24,7 +24,7 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         public override Task AnalyzeSyntaxAsync(Document document, Action<Diagnostic> addDiagnostic, CancellationToken cancellationToken)
         {
             // No syntax diagnostics produced by the EnC engine.  
-            return Task.FromResult(false);
+            return SpecializedTasks.EmptyTask;
```

Comment:
It looks like a non-equivalent replacement. Are we sure the return value of type `Task` is never downcast to `Task<bool>` by a caller?